lib/fibonacci.rb

```diff
@@ -1,8 +1,19 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+# Time Complexity - O(n)
+# Space Complexity - O(n) (should be O(n))
 # Hint, you may want a recursive helper method
 def fibonacci(n)
```

Comment:
This solution works, but you can actually solve it by only keeping the last two Fibonacci numbers rather than the whole array.